### PR TITLE
SCUMM: Fix encoding for Hebrew dialogs

### PR DESCRIPTION
**Pull request by orgads**
Like Pause dialog.

### CODE REVIEW

**Comment by orgads** [comment]
Before:
image
After:
image

**Comment by orgads** [comment]
Some games use DOS862 encoding. Will push a fixed revision soon.

File: engines/scumm/string.cpp [hunk 1]
```diff
@@ -2075,4 +2075,30 @@ void ScummEngine::translateText(const byte *text, byte *trans_buff) {
...
+	case Common::ZH_TWN:
+	case Common::ZH_CNA:
+		return Common::kWindows950;
+	case Common::RU_RUS:
+		return Common::kDos866;
+	case Common::HE_ISR:
+		switch (_game.id) {
+		case GID_LOOM:
+		case GID_ZAK:
+			fakeBidiString(reinterpret_cast<byte *>(message.begin()), true);
```
Comment (orgads) [inline engines/scumm/string.cpp:2094]: I know this is super-ugly (especially with the current function name). If you have better ways, let me know :)

Comment (bluegr) [reply engines/scumm/string.cpp:2094]: No please don't modify variables inside a getter. Split this bit into a separate function

Comment (orgads) [reply engines/scumm/string.cpp]: Ok, separated.

**Comment by orgads** [comment]
ping?

File: engines/scumm/string.cpp [hunk 2]
```diff
@@ -2075,4 +2075,40 @@ void ScummEngine::translateText(const byte *text, byte *trans_buff) {
 	memcpy(trans_buff, text, resStrLen(text) + 1);
 }
 
+bool ScummEngine::reverseIfNeeded(const byte *text, byte *reverseBuf) const
```
Comment (bluegr) [inline engines/scumm/string.cpp:2078]:
```suggestion
bool ScummEngine::reverseIfNeeded(const byte *text, byte *reverseBuf) const {
```

File: engines/scumm/string.cpp [hunk 3]
```diff
@@ -2075,4 +2075,40 @@ void ScummEngine::translateText(const byte *text, byte *trans_buff) {
...
+	if (_language != Common::HE_ISR)
+		return false;
+	if (_game.id != GID_LOOM && _game.id != GID_ZAK)
+		return false;
+	strcpy(reinterpret_cast<char *>(reverseBuf), reinterpret_cast<const char *>(text));
+	fakeBidiString(reverseBuf, true);
+	return true;
+}
+
+Common::CodePage ScummEngine::getDialogCodePage() const
```
Comment (bluegr) [inline engines/scumm/string.cpp:2089]:
```suggestion
Common::CodePage ScummEngine::getDialogCodePage() const {
```

File: engines/scumm/string.cpp [hunk 4]
```diff
@@ -2075,4 +2075,40 @@ void ScummEngine::translateText(const byte *text, byte *trans_buff) {
 	memcpy(trans_buff, text, resStrLen(text) + 1);
 }
 
+bool ScummEngine::reverseIfNeeded(const byte *text, byte *reverseBuf) const
+{
```
Comment (bluegr) [inline engines/scumm/string.cpp:2079]:
```suggestion
```

File: engines/scumm/string.cpp [hunk 5]
```diff
@@ -2075,4 +2075,40 @@ void ScummEngine::translateText(const byte *text, byte *trans_buff) {
...
+		return false;
+	if (_game.id != GID_LOOM && _game.id != GID_ZAK)
+		return false;
+	strcpy(reinterpret_cast<char *>(reverseBuf), reinterpret_cast<const char *>(text));
+	fakeBidiString(reverseBuf, true);
+	return true;
+}
+
+Common::CodePage ScummEngine::getDialogCodePage() const
+{
```
Comment (bluegr) [inline engines/scumm/string.cpp:2090]:
```suggestion
```

**Comment by bluegr** [comment]
Overall, this looks good now, thanks! Nice work, merging

**Pull request merged**